### ISSUE TITLE
Cannot change state in private window

### ISSUE THREAD

**Issue by TheDcoder** [raicem/impulse-blocker#34]
Changing the state of impulse blocker (On/Off) doesn't work in a private window. In issue #21 @raicem made a [comment](https://github.com/raicem/impulse-blocker/issues/21#issuecomment-439653893) regarding this problem:

> You are definitely right about the on/off options being unselected. This is due to the fact that we can't access the background process of the extension so we cannot get extension status, we cannot get the url, we cannot add or remove sites from the toolbar/popup. The workaround is cumbersome. Until that time I think I should add a warning about it.

This is a bit frustrating for me as I use private mode more often, sometimes I want to watch an YouTube video linked from another site in a private window (so that it doesn't get associated with my account) but only to find out that I forgot to turn off the impulse blocker.

I am creating this ticket to raise more awareness and for more proper bug tracking for this issue :smile:

**Comment by raicem** [comment]
Thank for raising the awareness. I'm glad you did it because I forgot about that issue. I have some good news though. 

I rewrote the extension and while doing so I started to connect to the background process via messages rather than getting the complete instance of the whole bacground script. This solved that problem. Now I tried it the rewritter version and I think it still words. 

I am hoping to release the new version as 1.0.0. As today it is completed with the new settings page with the current features. However I wanna add some new features that would make it more useable and practical. So in week I plan to release it. 

Many thanks again for reminding this issue.
